contracts/Vault.vy

```diff
@@ -201,6 +202,7 @@ event StrategyAddedToQueue:
 
 # NOTE: Track the total for overhead targeting purposes
 strategies: public(HashMap[address, StrategyParams])
+strategies_addresses: public(address[MAXIMUM_STRATEGIES])
```

Comment:
Instead of using an array, I was hoping it would be possible to update the aggregate metric `lockedProfit` based on some more advanced math, happening each time `report` is called.

I think #273 might actually be solving this with the change in `report` that takes what locked profit was and adds it to the new gains.

Reply:
The issue with that strategy is that some profit from a previous report will be locked for longer than it should.
If strategy A reports 300 USD gain it will be locked for 6 hours.
Then strategy B reports 150 USD gain 3 hours after A, 300 USD will be locked for another 6 hours.

What can happen if a lot of strategies keep on reporting gains and taking some of the previous report generated gains, as the price per share is considered without the locked funds, some will find opportunities to buy some shares when the locked value become interesting enough.